dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderTelemetry.h

```diff
@@ -0,0 +1,101 @@
+
+public:
+    DEFINE_EVENT_METHOD(LogBuildNotification)(
+        winrt::hresult hr) noexcept try
+    {
+        if (m_telemetryHelper.ShouldLogEvent())
+        {
+            TraceLoggingClassWriteMeasure(
+                "BuildNotification",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingBool(m_telemetryHelper.IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(m_telemetryHelper.GetAppName().c_str(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(LogButtonToString)(
+        winrt::hresult hr) noexcept try
+    {
+        if (m_telemetryHelper.ShouldLogEvent())
+        {
+            TraceLoggingClassWriteMeasure(
+                "ButtonToString",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingBool(m_telemetryHelper.IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(m_telemetryHelper.GetAppName().c_str(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(LogComboBoxToString)(
+        winrt::hresult hr) noexcept try
+    {
+        if (m_telemetryHelper.ShouldLogEvent())
+        {
+            TraceLoggingClassWriteMeasure(
+                "ComboBoxToString",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingBool(m_telemetryHelper.IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(m_telemetryHelper.GetAppName().c_str(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(LogProgressBarToString)(
+        winrt::hresult hr) noexcept try
+    {
+        if (m_telemetryHelper.ShouldLogEvent())
+        {
+            TraceLoggingClassWriteMeasure(
+                "ProgressBarToString",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingBool(m_telemetryHelper.IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(m_telemetryHelper.GetAppName().c_str(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(LogTextPropertiesToString)(
+        winrt::hresult hr) noexcept try
+    {
+        if (m_telemetryHelper.ShouldLogEvent())
+        {
+            TraceLoggingClassWriteMeasure(
+                "TextPropertiesToString",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
```

Comment:
what is this for?

Reply:
This is required for all WindowsAppSDK telemetry. It provides version number, and other info about the build.